File: common/src/main/java/generations/gg/generations/core/generationscore/network/packets/shop/S2CSyncPlayerMoneyHandler.java
package generations.gg.generations.core.generationscore.network.packets.shop;

import generations.gg.generations.core.generationscore.api.player.ClientPlayerMoney;
import generations.gg.generations.core.generationscore.network.ClientNetworkPacketHandler;

public class S2CSyncPlayerMoneyHandler implements ClientNetworkPacketHandler<S2CSyncPlayerMoneyPacket> {
    @Override
    public void handle(S2CSyncPlayerMoneyPacket packet) {
        ClientPlayerMoney.balance = packet.balance;
    }
}

File: src/common/main/kotlin/generations/gg/generations/core/generationscore/common/network/packets/shop/S2CSyncPlayerMoneyHandler.kt
package generations.gg.generations.core.generationscore.common.network.packets.shop

import com.cobblemon.mod.common.api.net.ClientNetworkPacketHandler
import generations.gg.generations.core.generationscore.common.api.player.ClientPlayerMoney
import net.minecraft.client.Minecraft

object S2CSyncPlayerMoneyHandler : ClientNetworkPacketHandler<S2CSyncPlayerMoneyPacket> {
    override fun handle(packet: S2CSyncPlayerMoneyPacket, minecraft: Minecraft) {
        ClientPlayerMoney.balance = packet.balance
    }
}
